docs/smart-contracts/guides/dapps/Terminal-UI-Guide.mdx

```diff
@@ -0,0 +1,1603 @@
+
+### Setting up Inquirer.js
+
+To start using Inquirer.js, you first need to install it via npm:
+
+```bash
+npm install inquirer
+```
+
+Then, import it into your project:
+
+```javascript
+import inquirer from "inquirer";
+```
+
+### Creating prompts
+
+Here are some common types of prompts you can use with Inquirer.js:
+
+#### 1. List prompt
+
+The list prompt allows the user to select one option from a list of choices.
+
+```javascript
+const { network } = await inquirer.prompt([
+  {
+    type: "list",
+    name: "network",
+    message: "Select the network:",
+    choices: ["testnet", "mainnet", "futurenet"],
+  },
+]);
+```
+
+#### 2. Input prompt
+
+The input prompt asks the user to enter a string value.
+
+```javascript
+const { mintAmount } = await inquirer.prompt([
+  {
+    type: "input",
+    name: "mintAmount",
+    message: "Enter the amount of LP tokens to mint:",
+    validate: (input) =>
+      !isNaN(input) && Number(input) > 0 ? true : "Invalid amount",
+  },
+]);
+```
+
+#### 3. Confirm prompt
+
+The confirm prompt asks the user to confirm a yes/no question.
+
+```javascript
+const { confirm } = await inquirer.prompt([
+  {
+    type: "confirm",
+    name: "confirm",
+    message: `Proceed with the transaction to mint LP tokens using ${estimatedBLND} BLND and ${estimatedUSDC}?`,
+  },
+]);
+```
+
+### Handling user input
+
+After defining the prompts, Inquirer.js returns a promise that resolves with an object containing the user's answers. You can then use these answers in your application logic.
+
+### Example usage
+
+Below is a step-by-step explanation of how the LP_CLI.ts file uses Inquirer.js to interact with the user:
+
+1. **Prompt the user for network selection**:
+
+   - The user is presented with a list of networks to choose from (`testnet`, `mainnet`, `futurenet`).
+   - The selected network is stored in the `network` variable.
+
+2. **Load the address book**:
+
+   - Based on the selected network, the address book is loaded, which contains the contract addresses.
+
+3. **Prompt the user for mint amount and slippage**:
+
+   - The user is asked to enter the amount of LP tokens to mint.
+   - The user is also asked to specify the maximum slippage percentage.
+
+4. **Mint LP tokens**:
+
+   - The `mintLPTokens` function is called with the address book, mint amount, and slippage.
+   - This function handles the logic of estimating the required BLND and USDC amounts, building the transaction, simulating it, and finally submitting it.
+
+5. **Transaction confirmation**:
+   - Before submitting the transaction, the user is asked to confirm if they want to proceed.
+   - If the user confirms, the transaction is signed and submitted.
+
+## Running the CLI
+
+### Install and build the project
+
+Run the following commands to install dependencies and build the project:
+
+```bash
+yarn install
+yarn run build
+```
+
+### Running the CLI
+
+Execute the following command to run the CLI:
+
+```bash
+yarn run CLI
```

Comment:
There is no script named cli, and the script in the origin repo is not capitalized.